src/metamodels/crosscat.py

```diff
@@ -220,15 +226,34 @@ class CrosscatMetamodel(metamodel.IBayesDBMetamodel):
 
     Internally, the Crosscat metamodel adds SQL tables to the database
     with names that begin with ``bayesdb_crosscat_``.
+
+    `crosscat` can be a crosscat engine instance or constructor. Prefer to pass
+    a constructor, since that way the engine obtains a random seed from the
+    BayesDB object. The 'crosscat as instance' case is kept for backwards
+    compatibility.
+
+    Arguments to the constructor can be passed by ccargs and cckwargs. If using
+    a crosscat MultiprocessingEngine, the process pool must be passed as a
+    'pool' argument via cckwargs. It would not do for a new process pool to be
+    created every time a new query is made.
+
     """
 
     def __init__(self, crosscat, subsample=None):
-        if subsample is None:
-            subsample = False
         self._crosscat = crosscat
-        self._subsample = subsample
+        self._subsample = False if subsample is None else subsample
         self._theta_validator = crosscat_theta_validator.Validator()
 
+    def _crosscat_engine(self, bdb):
+        if isinstance(self._crosscat, EngineTemplate):
```

Comment:
Concrete suggestion is to summarize what the intended change in interface is up front, in the pull request comment.  Too many conceptual changes in the control and data flow here to fit into my head in order to make a local suggestion, I'm afraid.